core/config/project_settings.cpp

```diff
@@ -1136,20 +1136,28 @@ Variant ProjectSettings::get_setting(const String &p_setting, const Variant &p_d
 	}
 }
 
-Array ProjectSettings::get_global_class_list() {
-	Array script_classes;
+TypedArray<Dictionary> ProjectSettings::get_global_class_list() {
+	static bool loaded = false;
```

Comment:
Why static local variable instead of a field in class?
Also right now the list can't be reloaded (which is important in editor). You could probably set the array in `store_global_class_list()`.

Reply:
I looked for some code examples about how that once-only init is done elsewhere in the engine source code and that pattern is used often. I guess it's a bit less mental burden as that static variable is only used inside one function, but I can easily change it if needed.

Reply:
Any example? I only found it in print methods, but that's not the same use.

Reply:
In `Object::initialize_class()` there is `static bool initialized`, but now that I look it's not really that common. I I think that was just the first example I found and assumed it's more common as that is pretty core code. I could just change that to non-static member variable like "bool global_class_list_loaded" (free to bikeshed the name)?

Reply:
Sound ok.